packages/at_secondary_server/lib/src/verb/handler/enroll_verb_handler.dart

```diff
@@ -203,20 +203,29 @@ class EnrollVerbHandler extends AbstractVerbHandler {
     }
     enrollStatus ??=
         getEnrollStatusFromString(enrollDataStoreValue!.approval!.state);
-    // Validates if enrollment is not expired
+
     if (EnrollStatus.expired == enrollStatus) {
       response.isError = true;
       response.errorCode = 'AT0028';
       response.errorMessage =
           'enrollment_id: $enrollmentIdFromParams is expired or invalid';
-    }
-    if (response.isError) {
       return;
     }
-    // Verifies whether the enrollment state matches the intended state
-    // Throws AtEnrollmentException, if the enrollment state is different from
-    // the intended state
-    _verifyEnrollmentStateBeforeAction(operation, enrollStatus);
+    // If the current operation is approve, but the enrollment is already approved
```

Comment:
if current operation is approve and enrollment is already approved, we should return with Enrollment already approved response

Reply:
addressed in [6cd81fa](https://github.com/atsign-foundation/at_server/pull/1562/commits/6cd81faef234f9eb77d2743856046f94e27720ee)